### PR TITLE
Various Dwell related bug fixes

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
This PR fixes the following bugs in the dwell feature/example:
- No profile error when entering play mode in editor
- The dwell timer does not stop after the object is no longer being focused on
- For the decay dwell example, the focus time was not calculated correctly after user looking away and then back at the object due to the "decayed time" was not accounted for

## Changes
- Fixes: #9168

### CODE REVIEW

File: Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs [hunk 1]
```diff
@@ -14,23 +14,11 @@ namespace Microsoft.MixedReality.Toolkit.Experimental.Dwell
     [Serializable]
     public class DwellProfileWithDecay : DwellProfile
     {
-        [Tooltip("Should the system allow for dwell to resume if the pointer exits the target briefly.")]
-        [SerializeField]
-        private bool allowDwellDecayOnCancel = false;
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs:19]: Is this a feature we want to support, it just wasn't working? Or is it something else?

@ritijain

Comment (MaxWang-MS) [reply Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs:19]: I do not believe that Boolean is meaningful in its current state - the profile only works when it is true and also conceptually I do not see the reason setting it to false (i.e. if you do not want to resume why using dwell with decay in the first place?)

Comment (ritijain) [reply Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs:19]: @keveleigh - for gaze based dwell there are many things needed to stabilize the user's gaze and account for unintended errors... 
This property is expected to provide one of those stabilizations where - you intend to dwell on a button, but your gaze wanders off for a bit (couple ms), and so this property was meant to add some resiliency to the system to not cancel the dwell right away. This is sufficient for head gaze based dwell that Guides has, but not for eye gaze based dwell.

This iteration of work did not contain any support for eye gaze stabilization - but it may be worth keeping this around since the eye gaze stabilization logic that Sophie wrote in her examples was not ported  here.

Comment (keveleigh) [reply Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs:19]: >it may be worth keeping this around since the eye gaze stabilization logic that Sophie wrote in her examples was not ported here.

@ritijain We do have _some_ smoothing for eye gaze from Sophie at https://github.com/microsoft/MixedRealityToolkit-Unity/blob/ab5c87a3c49478d05fd67d1ee8ed298a915c3f94/Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityEyeGazeDataProvider.cs#L220-L290, but it might be out of date at this point

Comment (MaxWang-MS) [reply Assets/MRTK/Examples/Common/Scripts/DwellProfileWithDecay.cs:19]: @ritijain @keveleigh I understand the need for stabilization/not cancel the dwell just because the cursor goes off for a short amount of time, but the file in question is dwell with decay, which means it does not seem to make sense if we would keep that field around as in this case the field should _always_ be true. If that field is set to false then the "decay" no longer makes sense.

**Comment by cre8ivepark** [comment]
Minor example scene polish with timing adjustments. Thinking about a visual polish in a separate PR.
2021-01-28 16_43_19-Unity 2018 4 28f1 -  PREVIEW PACKAGES IN USE  - DwellExample unity - MRTK - Univ

**Review by keveleigh — APPROVED** [review]

File: Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset [hunk 2]
```diff
@@ -13,7 +13,7 @@ MonoBehaviour:
...
   m_EditorClassIdentifier: 
   dwellTriggerPointerType: 4
-  dwellIntentDelay: 0
-  dwellStartDelay: 2
-  timeToCompleteDwell: 5
-  timeToAllowDwellResume: 1
+  dwellIntentDelay: 0.5
+  dwellStartDelay: 0.5
+  timeToCompleteDwell: 2
+  timeToAllowDwellResume: 2
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset:19]: Are these changes intended? It looks like a lot of the profiles are now identical with the default settings

**Review by keveleigh — CHANGES_REQUESTED** [review]

File: Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset (hunk 2 above)
Comment (MaxWang-MS) [reply Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset:19]: I think @cre8ivepark made the changes (please see the commit history) and I tend to believe his judgements.

Comment (cre8ivepark) [reply Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset:19]: @keveleigh Yes, I have updated them since they were too slow to be triggered.

Comment (keveleigh) [reply Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset:19]: That works! @MaxWang-MS can we de-dupe the profiles that are now identical? Looks like at least InstantDwellProfile and ToggleDwellProfile.

Comment (MaxWang-MS) [reply Assets/MRTK/Examples/Experimental/Dwell/InstantDwellProfile.asset:19]: @keveleigh I would propose that we keep those two as it might cause confusion when the user tries to modify the behavior of a button only to find out another unrelated button also changes behavior. Currently the names of the profiles also clearly state what they do.

**Review by keveleigh — APPROVED** [review]
